front/lib/resources/content_fragment_resource.ts

```diff
@@ -226,11 +226,22 @@ export class ContentFragmentResource extends BaseResource<ContentFragmentModel>
       contentFormat: "text",
     });
 
+    let fileSid: string | null = null;
+    let snippet: string | null = null;
+
+    if (this.fileId) {
+      const file = await FileResource.fetchByModelId(this.fileId);
+      fileSid = file?.sId ?? null;
+
+      // Note: For CSV files outputted by tools, we have a "snippet" version of the output with the first rows stored in GCP, maybe it's better than our "summary" snippet stored on File.
+      // Need more testing, for now we are using the "summary" snippet.
+      snippet = file?.snippet ?? null;
+    }
+
     return {
       id: message.id,
-      fileId: this.fileId
-        ? FileResource.modelIdToSId({ id: this.fileId, workspaceId: owner.id })
```

Comment:
very surprised `modelIdToSId` is not used anymore?

Reply:
It's because this.fileId is a modelID but fileId in the rendered version is an SID. Since we are loading the fileResource anyway, I have the sid from the fileResource.

Reply:
(yes, confusing...)

Reply:
I see